packages/storage/amplify_storage_s3_dart/lib/src/model/s3_storage_list_options.dart

```diff
@@ -0,0 +1,56 @@
+// Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import 'package:amplify_core/amplify_core.dart';
+
+/// {@template storage.amplify_storage_s3.list_options}
+/// The configurable parameters for Storage S3 plugin `list` requests.
+/// {@endtemplate}
+class S3StorageListOptions extends StorageListOptions {
+  /// {@macro storage.amplify_storage_s3.list_options}
+  factory S3StorageListOptions({
+    StorageAccessLevel storageAccessLevel = StorageAccessLevel.guest,
+    int pageSize = 1000,
+  }) {
+    return S3StorageListOptions._(
+      storageAccessLevel: storageAccessLevel,
+      pageSize: pageSize,
+    );
+  }
+
+  const S3StorageListOptions._({
+    super.storageAccessLevel = StorageAccessLevel.guest,
+    super.pageSize = 1000,
+    this.targetIdentityId,
+  });
+
+  /// {@macro storage.amplify_storage_s3.list_options}
+  ///
+  /// Use when list objects that belongs to other user (identified by
+  /// `targetIdentityId`) rather than the currently signed user.
+  factory S3StorageListOptions.withTargetIdentityId({
```

Comment:
Nice 👍 yeah, better to design an API that leads people down the right path than to throw errors because you let them do something they weren't supposed to

Reply:
What do you think about naming it something like `S3StorageListOptions.forIdentity`. `withTargetIdentityId` doesn't feel super descriptive.

Reply:
Thought about starting with "for", but the options also have other properties...
If starts with "for" can change the factory signature to 

```dart
  factory S3StorageListOptions.forIdentity(
    String targetIdentityId, {
    int pageSize = 1000,
  })
```

What do you think?

Reply:
Yeah, I like that better 👍